Source/arcadia/internal/nrepl_support.clj

```diff
@@ -0,0 +1,67 @@
+(ns arcadia.internal.nrepl-support
+  (:import [BList]
+           [BDictionary]))
+
+(defn complete-symbol [text]
+  (let [[ns prefix-str] (as-> text <>
+                          (symbol <>)
+                          [(some-> <> namespace symbol) (name <>)])
+        ns-to-check (if ns
+                      (or ((ns-aliases *ns*) ns) (find-ns ns))
+                      *ns*)
+        fn-candidate-list (when ns-to-check
+                            (if ns
+                              (map str (keys (ns-publics ns-to-check)))
+                              (map str (keys (ns-map ns-to-check)))))]
+    (into '() (comp (filter #(.StartsWith % prefix-str))
+                    (map #(if ns (str ns "/" %) %))
+                    (map #(-> {:candidate %
+                               :type "function"})))
+          (concat
+           fn-candidate-list))))
+
+(defn complete-namespace [text]
+  (let [[ns prefix-str] (as-> text <>
+                          (symbol <>)
+                          [(some-> <> namespace symbol) (name <>)])
+        ns-candidate-list (when-not ns
+                            (map (comp str ns-name) (all-ns)))]
+    (into '() (comp (filter #(.StartsWith % prefix-str))
+                    (map str)
+                    (map #(-> {:candidate %
+                               :type "namespace"})))
+          ns-candidate-list)))
+
+(defn complete-keyword [text]
+  (let [keyword-candidate-list
+        ;; NOTE: :_ is used here to get an instance to some keyword.
```

Comment:
No need for a placeholder keyword, the following should give you the underlying hashmap 

```clj
(def sym-key-map
  (-> clojure.lang.Keyword
      (.GetField "_symKeyMap" (enum-or BindingFlags/NonPublic BindingFlags/Static))
      (.GetValue nil)))
```

And you only need to do that once, you should be able to reference `sym-key-map` in the `complete-keyword` function and it should see new keywords as they get interned.